force-app/main/default/lwc/refundPayment/refundPayment.js

```diff
@@ -33,48 +32,57 @@ export default class refundPayment extends LightningElement {
         refundAmount,
         refundPaymentDate,
         refundProcessing,
-        refundPaymentErrorMessage
+        refundPaymentErrorMessage,
+        loadingMessage,
+        spinnerAltText
     });
     refundView;
     paymentAmount;
     paymentDate;
+    currencyCode;
 
-    @wire(getRecord, {
-        recordId: '$recordId',
-        fields: [PAYMENT_AMOUNT_FIELD, PAYMENT_DATE_FIELD]})
-    wiredRecord({ error, data }) {
-        if (error) {
-            this.hasError = true;
-            this.errorMessage = constructErrorMessage(error).detail;
-        } else if (data) {
-            this.paymentAmount = getFieldValue(data, PAYMENT_AMOUNT_FIELD);
-            this.paymentDate = getFieldValue(data, PAYMENT_DATE_FIELD);
-        }
+    @api set recordId(value) {
```

Comment:
Remove Wire record and use setter function on `recordId`. This way we guarantee the record Id will exist when making the init call